caddyhttp/push/handler.go

```diff
@@ -0,0 +1,72 @@
+package push
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/mholt/caddy/caddyhttp/httpserver"
+)
+
+func (h Middleware) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+
+	pusher, hasPusher := w.(http.Pusher)
+
+	// No Pusher, no cry
+	if !hasPusher {
+		return h.Next.ServeHTTP(w, r)
+	}
+
+	// This is request for the pushed resource - it should not be recursive
+	if _, exists := r.Header[pushHeader]; exists {
+		return h.Next.ServeHTTP(w, r)
+	}
+
+	// Serve file first
+	code, err := h.Next.ServeHTTP(w, r)
+
+	if flusher, ok := w.(http.Flusher); ok {
+		flusher.Flush()
+	}
+
+outer:
+	for _, rule := range h.Rules {
+		if httpserver.Path(r.URL.Path).Matches(rule.Path) {
+			for _, resource := range rule.Resources {
+				pushErr := pusher.Push(resource.Path, &http.PushOptions{
+					Method: resource.Method,
+					Header: resource.Header,
+				})
+				if pushErr != nil {
+					// If we cannot push (either not supported or concurrent streams are full - break)
+					break outer
+				}
+			}
+		}
+	}
+
+	headers := w.Header()
```

Comment:
This variable is only referenced once... maybe we don't need to declare this one?